tools/Customization/DevHome.Customization/Views/OptimizeDevDriveDialog.xaml

```diff
@@ -18,10 +18,14 @@
     SecondaryButtonText="Cancel">
     <StackPanel>
         <TextBlock Text="{x:Bind ViewModel.OptimizeDevDriveDialogDescription}" TextWrapping="Wrap"/>
-        <Grid Margin="0 10 0 0">
-            <TextBox HorizontalAlignment="Left" MinWidth="300" Grid.Column="0" x:Name="DirectoryPathTextBox" PlaceholderText="Choose a directory" Text="{x:Bind ViewModel.DirectoryPathTextBox , Mode=TwoWay, UpdateSourceTrigger=PropertyChanged}"/>
+        <Grid Margin="0 10 0 0" >
+            <Grid.ColumnDefinitions>
+                <ColumnDefinition />
+                <ColumnDefinition Width="Auto" />
+            </Grid.ColumnDefinitions>
+            <TextBox HorizontalAlignment="Stretch" x:Name="DirectoryPathTextBox" PlaceholderText="{x:Bind ViewModel.ChooseDirectoryPlaceholderText, Mode=TwoWay}" Text="{x:Bind ViewModel.DirectoryPathTextBox , Mode=TwoWay, UpdateSourceTrigger=PropertyChanged}" />

```

Comment:
Do you really want to two-way bind placeholder text? Usually the PlaceholderText never changes, but the Text does. Then you can make it much simpler by using the localized resource directly here in the xaml, and don't need an observable property on the ViewModel.
```suggestion
            <TextBox HorizontalAlignment="Stretch" x:Name="DirectoryPathTextBox" x:Uid="ChooseDirectory" Text="{x:Bind ViewModel.DirectoryPathTextBox , Mode=TwoWay, UpdateSourceTrigger=PropertyChanged}" />
```
And change the resw file to:
```
  <data name="ChooseDirectory.PlaceholderText" xml:space="preserve">
    <value>Choose a directory</value>
    <comment>Choose directory</comment>
  </data>
```